### PR TITLE
method to get Message by link

### PR DESCRIPTION
**Pull request by Joshix-1**
Helpful method to get messages by their link.

### CODE REVIEW

File: javacord-api/src/main/java/org/javacord/api/DiscordApi.java [hunk 1]
```diff
@@ -754,6 +756,24 @@ default AccountUpdater createAccountUpdater() {
...
+     * Gets a message by its link.
+     *
+     * @param link The link of the message.
+     * @return The message with the given link.
+     */
+    default CompletableFuture<Message> getMessageByLink(String link) {
+        Matcher matcher = DiscordRegexPattern.MESSAGE_LINK.matcher(link);
+
+        if (!matcher.matches()) {
+            return CompletableFuture.completedFuture(null);
```
Comment (Vampire) [inline javacord-api/src/main/java/org/javacord/api/DiscordApi.java:769]: Javacord does not return `null`, except for `Void`, it should either be `CompletableFuture<Optional<Message>>` or `Optional<CompletableFuture<Message>>` which we both have in the codebase already. I guess here `Optional<CompletableFuture<Message>>` is more appropriate as the text channel is resolved first.

**Comment by Vampire** [comment]
But what if you have a valid link, but the channel id does not match a text channel.
That is where previously `null` was given back in the future and that is not right.

**Comment by Bastian** [comment]
Ohh right, I didn't cosider this case.
In that case, we should keep the `Optional<CompletableFuture<Message>> getMessageByLink(String link)` method (but still add the `getCachedMessageByLink(...)` method and the `IllegalArgumentException` for invalid links).

**Comment by Bastian** [comment]
Thank you! 👍

**Pull request merged**